editor/animation_track_editor.cpp

```diff
@@ -7008,6 +7013,23 @@ void AnimationTrackEditor::_cleanup_animation(Ref<Animation> p_animation) {
 	_update_tracks();
 }
 
+void AnimationTrackEditor::_toggle_function_name() {
+	_redraw_tracks();
+}
+
+void AnimationTrackEditor::_check_method_exist() {
+	for (int i = 0; i < animation->get_track_count(); i++) {
+		if (animation->track_get_type(i) == Animation::TrackType::TYPE_METHOD) {
+			if (animation->track_get_key_count(i) > 0) {
+				function_name_toggler->set_disabled(false);
+				return;
+			}
+		}
+	}
+	function_name_toggler->set_disabled(true);
+	function_name_toggler->set_pressed(false);
+}
+
```

Comment:
```suggestion
```

I think it is not necessary. If you check each time an animation is added, doesn't that mean that the settings will change each time an animation is added?

Reply:
My premise with this was to polish it up so you couldn't click on an icon in an animation with no function track, but I get your reasoning and besides snap mode and bezier curve, the others don't have this kind of check.